### ISSUE TITLE
JavadocMethod: Javadoc not found for an interface method

### ISSUE THREAD

**Issue by pbludov** [checkstyle/checkstyle#5226]
http://checkstyle.sourceforge.net/config_javadoc.html#JavadocMethod

/var/tmp $ javac TestClass.java
(no errors)

/var/tmp $ cat TestClass.java

```
package test;

interface TestInterface
{
  /**
   * Javadoc
   * 
   * @return foo
   */
  java.awt
    .image.renderable
    .ContextualRenderedImageFactory methodName(); // Javadoc NOT found
}

abstract class TestClass implements TestInterface
{
  /**
   * Javadoc
   * 
   * @return foo
   */
  public abstract java.awt
    .image.renderable
    .ContextualRenderedImageFactory methodName(); // Javadoc found
}
```
/var/tmp $ cat config.xml
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
  "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
  <property name="charset" value="UTF-8"/>
  <module name="TreeWalker">
    <module name="JavadocMethodCheck"/>
  </module>
</module>
```

/var/tmp $ java -Duser.language=en -Duser.country=US -jar checkstyle-8.3-all.jar -c config.xml TestClass.java

Starting audit...
[ERROR] /var/tmp/TestClass.java:12:5: Missing a Javadoc comment. [JavadocMethod]
Audit done.
Checkstyle ends with 1 errors.

---------------

When the full package name is present in the returned method type, and the package name is too long and needs to be wrapped, JavadocMethodCheck can not find the javadoc comment.

This bug affects only interface methods, class methods are ok.

--------------

/var/tmp $ java -Duser.language=en -Duser.country=US -jar checkstyle-8.3-all.jar -T TestClass.java

```
PACKAGE_DEF -> package [1:0]
|--ANNOTATIONS -> ANNOTATIONS [1:8]
|--IDENT -> test [1:8]
`--SEMI -> ; [1:12]
INTERFACE_DEF -> INTERFACE_DEF [3:0]
|--MODIFIERS -> MODIFIERS [3:0]
|   `--LITERAL_PUBLIC -> public [3:0]
|--LITERAL_INTERFACE -> interface [3:7]
|--IDENT -> TestInterface [3:17]
`--OBJBLOCK -> OBJBLOCK [4:0]
    |--LCURLY -> { [4:0]
    |--METHOD_DEF -> METHOD_DEF [12:4]
    |   |--MODIFIERS -> MODIFIERS [12:4]
    |   |--TYPE -> TYPE [12:4]
    |   |   `--DOT -> . [12:4]
    |   |       |--DOT -> . [11:10]
    |   |       |   |--DOT -> . [11:4]
    |   |       |   |   |--DOT -> . [10:6]
    |   |       |   |   |   |--BLOCK_COMMENT_BEGIN -> /* [5:2]
    |   |       |   |   |   |   |--COMMENT_CONTENT -> *\n   * Javadoc\n   * \n   * @return foo\n    [5:4]
    |   |       |   |   |   |   `--BLOCK_COMMENT_END -> */ [9:3]
    |   |       |   |   |   |--IDENT -> java [10:2]
    |   |       |   |   |   `--IDENT -> awt [10:7]
    |   |       |   |   `--IDENT -> image [11:5]
    |   |       |   `--IDENT -> renderable [11:11]
    |   |       `--IDENT -> ContextualRenderedImageFactory [12:5]
    |   |--IDENT -> methodName [12:36]
    |   |--LPAREN -> ( [12:46]
    |   |--PARAMETERS -> PARAMETERS [12:47]
    |   |--RPAREN -> ) [12:47]
    |   `--SEMI -> ; [12:48]
    |--SINGLE_LINE_COMMENT -> // [12:50]
    |   `--COMMENT_CONTENT ->  Javadoc NOT found\n [12:52]
    `--RCURLY -> } [13:0]
CLASS_DEF -> CLASS_DEF [15:0]
|--MODIFIERS -> MODIFIERS [15:0]
|   |--LITERAL_PUBLIC -> public [15:0]
|   `--ABSTRACT -> abstract [15:7]
|--LITERAL_CLASS -> class [15:16]
|--IDENT -> TestClass [15:22]
|--IMPLEMENTS_CLAUSE -> implements [15:32]
|   `--IDENT -> TestInterface [15:43]
`--OBJBLOCK -> OBJBLOCK [16:0]
    |--LCURLY -> { [16:0]
    |--METHOD_DEF -> METHOD_DEF [22:2]
    |   |--MODIFIERS -> MODIFIERS [22:2]
    |   |   |--BLOCK_COMMENT_BEGIN -> /* [17:2]
    |   |   |   |--COMMENT_CONTENT -> *\n   * Javadoc\n   * \n   * @return foo\n    [17:4]
    |   |   |   `--BLOCK_COMMENT_END -> */ [21:3]
    |   |   |--LITERAL_PUBLIC -> public [22:2]
    |   |   `--ABSTRACT -> abstract [22:9]
    |   |--TYPE -> TYPE [24:4]
    |   |   `--DOT -> . [24:4]
    |   |       |--DOT -> . [23:10]
    |   |       |   |--DOT -> . [23:4]
    |   |       |   |   |--DOT -> . [22:22]
    |   |       |   |   |   |--IDENT -> java [22:18]
    |   |       |   |   |   `--IDENT -> awt [22:23]
    |   |       |   |   `--IDENT -> image [23:5]
    |   |       |   `--IDENT -> renderable [23:11]
    |   |       `--IDENT -> ContextualRenderedImageFactory [24:5]
    |   |--IDENT -> methodName [24:36]
    |   |--LPAREN -> ( [24:46]
    |   |--PARAMETERS -> PARAMETERS [24:47]
    |   |--RPAREN -> ) [24:47]
    |   `--SEMI -> ; [24:48]
    |--SINGLE_LINE_COMMENT -> // [24:50]
    |   `--COMMENT_CONTENT ->  Javadoc found\n [24:52]
    `--RCURLY -> } [25:0]
```

**Comment by pbludov** [comment]
As a dirty workaround it is possible to add the **abstract** modifier to the method:
```
interface TestInterface
{
  /**
   * Javadoc
   * 
   * @return foo
   */
  abstract java.awt
    .image.renderable
    .ContextualRenderedImageFactory methodName(); // Javadoc found
}
```

**Comment by pbludov** [comment]
```
class TestClass
{
  /**
   * Javadoc
   * 
   * @return foo
   */
  java.awt
    .image.renderable
    .ContextualRenderedImageFactory methodName() {
      return null;
    }
}
```
Also fails. The modifiers are mondatory to find a javadoc comment.

**Comment by romani** [comment]
@pbludov , please beware that JavadocMethod is not ANTLR based Check so it is very fragile. We need to reimplement it completely to ANTLR based validation and probably redesign it as it try to validate too much.

**Comment by MihneaS** [comment]
I have a problem that seems the same to me with DesignForExtension (also from TreeWalker), it would not recognize any javadoc above a method with no modifiers.

**Comment by romani** [comment]
@MihneaS , please create separate issue on your problem, DesignForExtension  and JavadocMethod are unrelated implementations.

**Comment by pbludov** [comment]
Fixed as a part of Issue #4169
